### PR TITLE
[doc] Add docs of node service

### LINKED ISSUE #162: [kwok] Simulate the services provided by Kubelet
### What would you like to be added?

Simulate the services provided by Kubelet for `logs`, `exec`, `attach`, `port-forward`.

- [x] #198 #238
- [x] #236 #430 #446
- [x] #216 #233
- [x] #217 #417
- [x] #237

### Why is this needed?

#159

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:

Fixes #162

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by wzshiming** [comment]
/cc @nikola-jokic
/cc @windsonsea

I have prepared documentation for the mock node service, as the functionality is not quite finished yet, it could help to see if there are any problems with the logic and functional description, could you have time to review it?

File: demo/exec.demo [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Let's getting started with kwokctl!
```
Comment (windsonsea) [inline demo/exec.demo:1]:
```suggestion
# Let's get started with kwokctl!
```

File: demo/port-forward.demo [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Let's getting started with kwokctl!
```
Comment (windsonsea) [inline demo/port-forward.demo:1]:
```suggestion
# Let's get started with kwokctl!
```

File: demo/port-forward.demo [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+kwokctl create cluster -c ./test/kwokctl/port-forward.yaml
+
+# Create a node.
+kubectl apply -f ./test/kwok/fake-node.yaml
+
+# Apply a deployment.
+kubectl apply -f ./test/kwok/fake-deployment.yaml
+
+# Port forward to the deployment.
```
Comment (windsonsea) [inline demo/port-forward.demo:10]:
```suggestion
# Forward port to the deployment.
```
This is only for consistency style: verb + noun

File: site/content/en/docs/user/attach-configuration.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+{{< hint "info" >}}
+
+This document walks you through how to configure the Attach feature.
+
+{{< /hint >}}
+
+## What is a Attach?
+
+The Attach API is a [`kwok` Configuration]({{< relref "/docs/user/configuration" >}}) that allows users to define and simulate attach to an Pod(s).
```
Comment (windsonsea) [inline site/content/en/docs/user/attach-configuration.md:11]:
```suggestion
The Attach API is a [`kwok` Configuration]({{< relref "/docs/user/configuration" >}}) that allows users to define and simulate attaching to Pod(s).
```

File: site/content/en/docs/user/exec-configuration.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+{{< hint "info" >}}
+
+This document walks you through how to configure the Exec feature.
+
+{{< /hint >}}
+
+## What is a Exec?
+
+The Exec API is a [`kwok` Configuration]({{< relref "/docs/user/configuration" >}}) that allows users to define and simulate exec to an Pod(s).
```
Comment (windsonsea) [inline site/content/en/docs/user/exec-configuration.md:11]:
```suggestion
The Exec API is a [`kwok` Configuration]({{< relref "/docs/user/configuration" >}}) that allows users to define and simulate exec to Pod(s).
```

File: site/content/en/docs/user/logs-configuration.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+{{< hint "info" >}}
+
+This document walks you through how to configure the Logs feature.
+
+{{< /hint >}}
+
+## What is a Logs?
+
+The Logs API is a [`kwok` Configuration]({{< relref "/docs/user/configuration" >}}) that allows users to define and simulate logs to an Pod(s).
```
Comment (windsonsea) [inline site/content/en/docs/user/logs-configuration.md:11]:
```suggestion
The Logs API is a [`kwok` Configuration]({{< relref "/docs/user/configuration" >}}) that allows users to define and simulate logs to Pod(s).
```

File: site/content/en/docs/user/port-forward-configuration.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+{{< hint "info" >}}
+
+This document walks you through how to configure the PortForward feature.
+
+{{< /hint >}}
+
+## What is a PortForward?
+
+The PortForward API is a [`kwok` Configuration]({{< relref "/docs/user/configuration" >}}) that allows users to define and simulate port forwarding to an Pod(s).
```
Comment (windsonsea) [inline site/content/en/docs/user/port-forward-configuration.md:11]:
```suggestion
The PortForward API is a [`kwok` Configuration]({{< relref "/docs/user/configuration" >}}) that allows users to define and simulate port forwarding to Pod(s).
```

**Review by windsonsea — COMMENTED** [review]
only some grammar issues

File: site/content/en/docs/user/attach-configuration.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  name: <string>
+  namespace: <string>
+spec:
+  attaches:
+    - containers:
+        - <string>
+      logsFile: <string>
+```
+
+To attach a container, you can set the `attaches` field in the spec section of a Logs resource.
```
Comment (nikola-jokic) [inline site/content/en/docs/user/attach-configuration.md:28]:
```suggestion
To attach a container, you can set the `attaches` field in the spec section of an Attach resource.
```

**Comment by windsonsea** [comment]
Let's kick this in except it has some `good first issues` that can attract contributors.

/lgtm

File: site/content/en/docs/user/attach-configuration.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  namespace: <string>
+spec:
+  attaches:
+    - containers:
+        - <string>
+      logsFile: <string>
+```
+
+To attach a container, you can set the `attaches` field in the spec section of an Attach resource.
+The `containers` field is used to match an item in the `attaches` field. If the `containers` field is not set, the `attaches` item will default to all containers.
```
Comment (nikola-jokic) [inline site/content/en/docs/user/attach-configuration.md:29]: The only thing I would add here is what does it mean to attach to all containers? Kubectl will default to the first one if no annotation is being set. Should we replicate that logic if no containers are specified?

Comment (wzshiming) [reply site/content/en/docs/user/attach-configuration.md]: I got lost in notifications and am now updated.

**Pull request merged**